docs/content/latest/explore/ysql-language-features/data-manipulation.md

```diff
@@ -0,0 +1,343 @@
+YSQL provides the `INSERT ON CONFLICT` statement that you can use to perform upserts, as follows:
+
+```sql
+INSERT INTO employees (employee_no, name, department) 
+VALUES (1, 'John Smith', 'Sales')
+ON CONFLICT (department) 
+DO NOTHING;
+```
+
+## Loading Data from a File
+
+The `COPY FROM` statement allows you to populate a table by loading data from a file whose columns are separated by a delimiter character. If the table already has data, the `COPY FROM` statement appends the new data to the existing data by creating new rows. Table columns that are not specified in the `COPY FROM` column list are populated with their default values.
+
+The `filename` parameter of the `COPY` statement enables reading directly from a file.
+
+The following example demonstrates how to use the `COPY FROM` statement:
+
+```sql
+COPY employees FROM '/home/mydir/employees.txt.sql' DELIMITER ',' CSV HEADER;
+```
+## Exporting Data to a File
+
+The `COPY TO` statement allows you to export data from a table to a file. By specifying a column list, you can instruct `COPY TO` to only export data from certain columns.
+
+The `filename` parameter of the `COPY` statement enables copying to a file directly.
+
+The following example demonstrates how to use the `COPY FROM` statement:
+
+```sql
+COPY employees TO '/home/mydir/employees.txt.sql' DELIMITER ',';
+```
+
+## Backing Up a Database
+
+You can back up a single instance of a YugabyteDB database into a plain-text SQL file by using the  `ysql_dump` is a utility, as follows:
+
+```
+ysql_dump mydb > mydb.sql
+```
+
+To back up global objects that are common to all databases in a cluster, such as roles, you need to use  `ysql_dumpall` .
+
+`ysql_dump` makes backups regardless of whether or not the database is being used.
+
+To reconstruct the database from a plain-text SQL file to the state the database was in at the time of saving, import this file using the `ysqlsh \i`  command, as follows:
+
+```
+ysqlsh \i mydb
+```
+
+## Configuring Automatic Timestamps
+
+You can use automatic timestamps to keep track of when data in a table was added or updated. 
+
+The date of the data creation is typically added via a  `created_at` column with a default value of `NOW()`, as shown in the following example: 
+
+```sql
+CREATE TABLE employees (
+    employee_no integer NOT NULL,
+    name text,
+    department text,
+  	created_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+```
+
+To track updates, you need to use triggers that let you define functions executed when an update is performed. The following example shows how to create a function in PL/pgSQL that returns an object called `NEW` containing data being modified:
+
+```sql
+CREATE OR REPLACE FUNCTION trigger_timestamp()
+RETURNS TRIGGER AS $$
+BEGIN
+  NEW.updated_at = NOW();
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+```
+
+The following examples create a table and connect it with a trigger that executes the `trigger_timestamp` function every time a row is updated in the table:
+
```

Comment:
Since the reader following along may not notice the small delta of `created_at` vs `updated_at` you could either add in `DROP TABLE employees` here, or you could

```
yugabyte=# ALTER TABLE employees ADD COLUMN updated_at TIMESTAMP NOT NULL DEFAULT NOW();
ALTER TABLE
yugabyte=# \d employees 
                          Table "public.employees"
   Column    |            Type             | Collation | Nullable | Default 
-------------+-----------------------------+-----------+----------+---------
 employee_no | integer                     |           | not null | 
 name        | text                        |           |          | 
 department  | text                        |           |          | 
 created_at  | timestamp without time zone |           | not null | now()
 updated_at  | timestamp without time zone |           | not null | now()
# select * from employees ;
 employee_no |    name    | department |         created_at         | updated_at 
-------------+------------+------------+----------------------------+------------
           4 | Desi Arnaz | Bandleader | 2021-01-28 05:50:04.488208 | 
```

and then after the create trigger

```CREATE TRIGGER
yugabyte=# INSERT INTO employees (employee_no, name, department ) VALUES (5, 'Eliot Daniel', 'Composer');
INSERT 0 1
yugabyte=# select * from employees ;
 employee_no |     name     | department |         created_at         |         updated_at         
-------------+--------------+------------+----------------------------+----------------------------
           5 | Eliot Daniel | Composer   | 2021-01-28 05:59:38.049836 | 2021-01-28 05:59:38.049836
           4 | Desi Arnaz   | Bandleader | 2021-01-28 05:50:04.488208 | 
(2 rows)
```

Reply:
Thanks for the suggestion.